RecoLocalTracker/SiPixelRecHits/src/PixelCPEGenericBase.cc

```diff
@@ -0,0 +1,237 @@
+    bool& edgey,
+    bool& bigInX,
+    bool& bigInY,
+    int& maxPixelCol,
+    int& maxPixelRow,
+    int& minPixelCol,
+    int& minPixelRow,
+    uint& sizex,
+    uint& sizey,
+    DetParam const& theDetParam,
+    ClusterParamGeneric const& theClusterParam) const {  // Default errors are the maximum error used for edge clusters.
+  // These are determined by looking at residuals for edge clusters
+  xerr = edgeClusterErrorX_ * micronsToCm;
+  yerr = edgeClusterErrorY_ * micronsToCm;
+
+  // Find if cluster is at the module edge.
+  maxPixelCol = theClusterParam.theCluster->maxPixelCol();
+  maxPixelRow = theClusterParam.theCluster->maxPixelRow();
+  minPixelCol = theClusterParam.theCluster->minPixelCol();
+  minPixelRow = theClusterParam.theCluster->minPixelRow();
+
+  edgex = (theDetParam.theRecTopol->isItEdgePixelInX(minPixelRow)) ||
+          (theDetParam.theRecTopol->isItEdgePixelInX(maxPixelRow));
+  edgey = (theDetParam.theRecTopol->isItEdgePixelInY(minPixelCol)) ||
+          (theDetParam.theRecTopol->isItEdgePixelInY(maxPixelCol));
+
+  sizex = theClusterParam.theCluster->sizeX();
+  sizey = theClusterParam.theCluster->sizeY();
+
+  // Find if cluster contains double (big) pixels.
+  bigInX = theDetParam.theRecTopol->containsBigPixelInX(minPixelRow, maxPixelRow);
+  bigInY = theDetParam.theRecTopol->containsBigPixelInY(minPixelCol, maxPixelCol);
+};
+
+void PixelCPEGenericBase::setXYErrors(float& xerr,
+                                      float& yerr,
+                                      const bool edgex,
+                                      const bool edgey,
+                                      const unsigned int sizex,
+                                      const unsigned int sizey,
+                                      const bool bigInX,
+                                      const bool bigInY,
+                                      const bool useTemplateErrors,
+                                      DetParam const& theDetParam,
+                                      ClusterParamGeneric const& theClusterParam) const {
+  if (useTemplateErrors) {
+    if (!edgex) {  // Only use this for non-edge clusters
+      if (sizex == 1) {
+        if (!bigInX) {
+          xerr = theClusterParam.sx1;
+        } else {
+          xerr = theClusterParam.sx2;
+        }
+      } else {
+        xerr = theClusterParam.sigmax;
+      }
+    }
+
+    if (!edgey) {  // Only use for non-edge clusters
+      if (sizey == 1) {
+        if (!bigInY) {
+          yerr = theClusterParam.sy1;
+        } else {
+          yerr = theClusterParam.sy2;
+        }
+      } else {
+        yerr = theClusterParam.sigmay;
+      }
+    }
+
+    // if (localPrint) {
+    //   cout << " in if " << edgex << " " << edgey << " " << sizex << " " << sizey << endl;
+    //   cout << " errors  " << xerr << " " << yerr << " " << theClusterParam.sx1 << " " << theClusterParam.sx2 << " "
+    //        << theClusterParam.sigmax << endl;  //dk
+    // }
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant

Reply:
removed